src/Plugin/EntityReferenceSelection/OgSelection.php

```diff
@@ -78,7 +104,7 @@ protected function buildEntityQuery($match = NULL, $match_operator = 'CONTAINS')
 
     $ids = [];
 
-    if ($this->configuration['handler_settings']['field_mode'] == 'admin') {
+    if (empty($this->configuration['handler_settings']['field_mode'])) {
```

Comment:
> Checking for empty() like this on a setting is asking for more bugs, I think we should keep explicit checking for types, if it's not set for whatever reason, we want to know about it

yeah, fully agree. no baby sitting code. Anyway, as discussed the field mode should be moved outside of the field config.